Source: https://lists.suckless.org/hackers/1801/15783.html
diff --git a/config.def.h b/config.def.h
index 50b40a4..18e179c 100644
--- a/config.def.h
+++ b/config.def.h
@@ -129,7 +129,9 @@ static unsigned int defaultrcs = 257;
 
 /*
  * Default shape of cursor
+ * 1: Blinking Block ("█")
  * 2: Block ("█")
+ * 3: Blinking Underline ("_")
  * 4: Underline ("_")
  * 6: Bar ("|")
  * 7: Snowman ("☃")
diff --git a/x.c b/x.c
index ea4f35c..77e3761 100644
--- a/x.c
+++ b/x.c
@@ -1466,11 +1466,17 @@ xdrawcursor(int cx, int cy, Glyph g, int ox, int oy, Glyph og)
 		case 7: /* st extension: snowman (U+2603) */
 			g.u = 0x2603;
 		case 0: /* Blinking Block */
+		  if (IS_SET(MODE_BLINK))
+		  	break;
 		case 1: /* Blinking Block (Default) */
+		  if (IS_SET(MODE_BLINK))
+		  	break;
 		case 2: /* Steady Block */
 			xdrawglyph(g, cx, cy);
 			break;
 		case 3: /* Blinking Underline */
+		  if (IS_SET(MODE_BLINK))
+		  	break;
 		case 4: /* Steady Underline */
 			XftDrawRect(xw.draw, &drawcol,
 					borderpx + cx * win.cw,
@@ -1479,6 +1485,8 @@ xdrawcursor(int cx, int cy, Glyph g, int ox, int oy, Glyph og)
 					win.cw, cursorthickness);
 			break;
 		case 5: /* Blinking bar */
+		  if (IS_SET(MODE_BLINK))
+			break;
 		case 6: /* Steady bar */
 			XftDrawRect(xw.draw, &drawcol,
 					borderpx + cx * win.cw,
@@ -1827,11 +1835,9 @@ run(void)
 		}
 		if (FD_ISSET(ttyfd, &rfd)) {
 			ttyread();
-			if (blinktimeout) {
-				blinkset = tattrset(ATTR_BLINK);
-				if (!blinkset)
-					MODBIT(win.mode, 0, MODE_BLINK);
-			}
+			blinkset = blinktimeout || tattrset(ATTR_BLINK);
+			if (!blinkset)
+			  MODBIT(win.mode, 0, MODE_BLINK);
 		}
 
 		if (FD_ISSET(xfd, &rfd))
@@ -1860,8 +1866,11 @@ run(void)
 				XNextEvent(xw.dpy, &ev);
 				if (XFilterEvent(&ev, None))
 					continue;
-				if (handler[ev.type])
-					(handler[ev.type])(&ev);
+				if (handler[ev.type]) {
+				  	(handler[ev.type])(&ev);
+					lastblink = now;
+					if (IS_SET(MODE_BLINK)) MODBIT(win.mode, 0, MODE_BLINK);
+				}				
 			}
 
 			draw();
